watertap/unit_models/coag_floc_model.py

```diff
@@ -524,6 +630,173 @@ def eq_mass_transfer_term(self, t, p, j):
             else:
                 return self.control_volume.mass_transfer_term[t, p, j] == 0.0
 
+        # Constraint for the volume of each rapid mixing basin in series
+        #   Mines, R.O., Environmental engineering: Principles
+        #       and Practice, 1st Ed, John Wiley & Sons, 2014.
+        #       Ch. 6.
+        self.rapid_mixing_basin_vol = Var(
+            self.flowsheet().config.time,
+            initialize=1,
+            bounds=(0, 1000),
+            domain=NonNegativeReals,
+            units=units_meta("length") ** 3,
+            doc="Volume of each rapid mixing basin in the series",
+        )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Constraint for the volume of each rapid mixing basin",
+        )
+        def eq_rapid_mixing_basin_vol(self, t):
+            flow_rate = pyunits.convert(
+                self.control_volume.properties_in[t].flow_vol_phase["Liq"],
+                to_units=units_meta("length") ** 3 / pyunits.s,
+            )
+
+            return (
+                self.rapid_mixing_basin_vol[t]
+                == flow_rate * self.rapid_mixing_retention_time[t]
+            )
+
+        # Constraint for the power usage of the rapid mixers
+        #   Mines, R.O., Environmental engineering: Principles
+        #       and Practice, 1st Ed, John Wiley & Sons, 2014.
+        #       Ch. 6.
+        self.rapid_mixing_power = Var(
+            self.flowsheet().config.time,
+            initialize=0.01,
+            bounds=(0, 100),
+            domain=NonNegativeReals,
+            units=pyunits.kW,
+            doc="Power usage of the rapid mixing basins in kW",
+        )
+
+        @self.Constraint(
+            self.flowsheet().config.time,
+            doc="Constraint for the power usage of the rapid mixing basins",
+        )
+        def eq_rapid_mixing_power(self, t):
+            vel_grad = pyunits.convert(
+                self.rapid_mixing_vel_grad[t], to_units=units_meta("time") ** -1
+            )
```

Comment:
Is this unit converting  necessary?  I think the two are both `s ** -1` ?

Reply:
I have a general question on the use of `pyunits.convert`.  It looks to me that this command forces the unit change of a pyomo var or param.  If so, I would wonder why this is needed.  If an equation is well formulated, the unit must be balanced on two sides.  If a unit conversion is needed, does that suggest any issue with the equation?

Reply:
Yes, this is always needed. This is because the `units_meta` comes from the property package, which the user may define (if using a generic package). The variable `rapid_mixing_vel_grad` is defined by the developer in the unit model and is defined to have specific units (in this case `s**-1`). In order to ENSURE that the units are consistent between the user defined property package and the developer defined unit model variables, we should always convert our defined units into the units defined in `units_meta`. 

For instance, with this conversion, if the user defines the units of time for the flowsheet/properties to be in `min`, then the conversion with apply the appropriate factor to `rapid_mixing_vel_grad` to the `min` time basis before constructing the constraint. 

In this particular case, both are already in `s**-1`, but we cannot assume that will always be the case.

Reply:
@lbibl `pyunits.convert` does not change the defined units of the `Var` or `Param`. Instead, it determines the necessary unit conversion factor and adds it to the expression under construction. E.g.:

```
m = ConcreteModel()

m.v = Var(units=pyunits.kg)

print(pyunits.convert(m.v, to_units=pyunits.pound)

>>> (2.20462*pounds/kg)*m.v
```